toolkit/tools/internal/network/network_test.go

```diff
@@ -139,9 +139,7 @@ func TestDownloadFile(t *testing.T) {
 			dstDir := t.TempDir()
```

Comment:
https://github.com/microsoft/azurelinux/pull/9307#discussion_r1626770961 We should add a test here to check early cancel propagates correctly.